JavaBytecodeCompiler/src/main/java/org/openzen/zenscript/javabytecode/compiler/ArrayHelperType.java

```diff
@@ -0,0 +1,47 @@
+package org.openzen.zenscript.javabytecode.compiler;
+
+import org.objectweb.asm.Type;
+import org.openzen.zenscript.codemodel.type.ArrayTypeID;
+import org.openzen.zenscript.codemodel.type.GenericTypeID;
+import org.openzen.zenscript.codemodel.type.TypeID;
+import org.openzen.zenscript.javabytecode.JavaBytecodeContext;
+import org.openzen.zenscript.javashared.JavaClass;
+import org.openzen.zenscript.javashared.JavaContext;
+import org.openzen.zenscript.javashared.JavaMethod;
+
+public class ArrayHelperType {
+
+	private final TypeID elementType;
+	private final JavaBytecodeContext context;
+
+	public ArrayHelperType(TypeID elementType, JavaBytecodeContext context) {
+		this.elementType = elementType;
+		this.context = context;
+	}
+
+
+	public ArrayHelperType getWithOneDimensionLess() {
+		ArrayTypeID arrayTypeID = ((ArrayTypeID) elementType);
+		return new ArrayHelperType(arrayTypeID.removeOneDimension(), context);
+	}
+
+	public void newArray(JavaWriter javaWriter) {
+		if(elementType instanceof GenericTypeID) {
+
+			elementType.accept(javaWriter, new JavaTypeExpressionVisitor(context));
+			javaWriter.swap();
+			final JavaClass arrayClass = JavaClass.fromInternalName("java/lang/reflect/Array", JavaClass.Kind.CLASS);
+			javaWriter.invokeStatic(JavaMethod.getStatic(arrayClass, "newInstance", "(Ljava/lang/Class;I)Ljava/lang/Object;", 0));
+			javaWriter.checkCast("[Ljava/lang/Object;");
+		} else {
+			javaWriter.newArray(context.getType(elementType));
+		}
+	}
+
+	public Type getASMType() {
+		if(CompilerUtils.isPrimitive(elementType)) {
+			return context.getType(elementType);
+		}
+		return Type.getType("[" + context.getDescriptor(elementType));
```

Comment:
I'm not too keen on the way this function is used atm.

Currently in one case it returns an array type and sometimes a primitive.
That looks to me as though back then I did screw up somewhere with the types and when these methods are called 🤔 


We could always return `context.getType(elementType)` here (and maybe rename it to getElementTypeAsASM or something).
Then, in ArrayInitializerHelper.java#195:
```diff
@@ -192,7 +192,7 @@ class ArrayInitializerHelper {
                } else {
                        visitMultiDimArray(javaWriter, sizeLocations, counterLocations, dim - 1, elementType, innermostFunction);
                }
-               javaWriter.arrayStore(elementType.getASMType());
+               javaWriter.arrayStore(currentArrayType.getASMType());
 
                //Return to the start
                javaWriter.iinc(forLoopCounter);
```

That way we would be consistent in always using the element type here..?